service/r2/store/kv/store.go

```diff
@@ -128,82 +130,115 @@ func (s *store) CreateNamespace(namespaceID string, uOpts r2.UpdateOptions) (*mo
 	// Get the latest view of the namespace.
 	view, err := ns.NamespaceView(len(ns.Snapshots()) - 1)
 	if err != nil {
-		return nil, s.handleUpstreamError(err)
+		return nil, handleUpstreamError(err)
 	}
 
 	return view, nil
 }
 
-func (s *store) DeleteNamespace(namespaceID string, uOpts r2.UpdateOptions) error {
+func (s *store) DeleteNamespace(namespaceID string, uOpts r2store.UpdateOptions) error {
 	nss, err := s.ruleStore.ReadNamespaces()
 	if err != nil {
-		return s.handleUpstreamError(err)
+		return handleUpstreamError(err)
 	}
 
 	rs, err := s.ruleStore.ReadRuleSet(namespaceID)
 	if err != nil {
-		return s.handleUpstreamError(err)
+		return handleUpstreamError(err)
 	}
 
 	meta := s.newUpdateMeta(uOpts)
 	err = nss.DeleteNamespace(namespaceID, rs.Version(), meta)
 	if err != nil {
-		return s.handleUpstreamError(err)
+		return handleUpstreamError(err)
 	}
 
 	mutable := rs.ToMutableRuleSet().Clone()
 	err = mutable.Delete(meta)
 	if err != nil {
-		return s.handleUpstreamError(err)
+		return handleUpstreamError(err)
 	}
 
 	if err = s.ruleStore.WriteAll(nss, mutable); err != nil {
-		return s.handleUpstreamError(err)
+		return handleUpstreamError(err)
 	}
 
 	return nil
 }
 
-func (s *store) FetchRuleSet(namespaceID string) (*models.RuleSetSnapshotView, error) {
+func (s *store) FetchRuleSetSnapshot(namespaceID string) (*models.RuleSetSnapshotView, error) {
 	rs, err := s.ruleStore.ReadRuleSet(namespaceID)
 	if err != nil {
-		return nil, s.handleUpstreamError(err)
+		return nil, handleUpstreamError(err)
 	}
 	return rs.Latest()
 }
 
 func (s *store) FetchMappingRule(namespaceID string, mappingRuleID string) (*models.MappingRuleView, error) {
-	crs, err := s.FetchRuleSet(namespaceID)
+	crs, err := s.FetchRuleSetSnapshot(namespaceID)
 	if err != nil {
 		return nil, err
 	}
 
 	mr, exists := crs.MappingRules[mappingRuleID]
 	if !exists {
-		return nil, s.mappingRuleNotFoundError(namespaceID, mappingRuleID)
+		return nil, mappingRuleNotFoundError(namespaceID, mappingRuleID)
 	}
 	return mr, nil
 }
 
+func (s *store) UpdateRuleSet(
```

Comment:
Can we move this to `ValidateRuleSet`? Grouping similar methods together makes it easier to navigate.